packages/features/ee/round-robin/roundRobinReassignment.ts

```diff
@@ -0,0 +1,365 @@
+    },
+    roundRobinReassignLogger
+  );
+
+  const reassignedRRHost = await getLuckyUser("MAXIMIZE_AVAILABILITY", {
+    availableUsers,
+    eventTypeId: eventTypeId,
+  });
+  const hasOrganizerChanged = booking.userId === previousRRHost.id;
+  const organizer = hasOrganizerChanged ? reassignedRRHost : booking.user;
+  const organizerT = await getTranslation(organizer?.locale || "en", "common");
+
+  if (hasOrganizerChanged) {
+    booking = await prisma.booking.update({
+      where: {
+        id: bookingId,
+      },
+      data: {
+        userId: reassignedRRHost.id,
+      },
+      select: bookingSelect,
+    });
+  } else {
+    const previousRRHostAttendee = booking.attendees.find(
+      (attendee) => attendee.email === previousRRHost.email
+    );
+    await prisma.attendee.update({
+      where: {
+        id: previousRRHostAttendee!.id,
+      },
+      data: {
+        name: reassignedRRHost.name || "",
+        email: reassignedRRHost.email,
+        timeZone: reassignedRRHost.timeZone,
+        locale: reassignedRRHost.locale,
+      },
+    });
+  }
+
+  const reassignedRRHostT = await getTranslation(reassignedRRHost.locale || "en", "common");
+
+  const teamMemberPromises = [];
+  for (const teamMember of eventType.hosts) {
+    const user = teamMember.user;
+    // Need to skip over the reassigned user and the organizer user
+    if (user.email === previousRRHost.email || user.email === organizer.email) {
+      continue;
+    }
+
+    if (attendeeEmailsSet.has(user.email)) {
+      const tTeamMember = await getTranslation(user.locale ?? "en", "common");
+
+      teamMemberPromises.push({
+        id: user.id,
+        email: user.email,
+        name: user.name || "",
+        timeZone: user.timeZone,
+        language: { translate: tTeamMember, locale: user.locale ?? "en" },
+      });
+    }
+  }
+
+  const teamMembers = await Promise.all(teamMemberPromises);
+  // Assume the RR host was labelled as a team member
+  if (reassignedRRHost.email !== organizer.email) {
+    teamMembers.push({
+      id: reassignedRRHost.id,
+      email: reassignedRRHost.email,
+      name: reassignedRRHost.name || "",
+      timeZone: reassignedRRHost.timeZone,
+      language: { translate: reassignedRRHostT, locale: reassignedRRHost.locale ?? "en" },
+    });
+  }
+
+  const attendeePromises = [];
+  for (const attendee of booking.attendees) {
+    if (
+      attendee.email === reassignedRRHost.email ||
+      attendee.email === previousRRHost.email ||
+      teamMembers.some((member) => member.email === attendee.email)
+    ) {
+      continue;
+    }
+    const tAttendee = await getTranslation(attendee.locale ?? "en", "common");
+
+    attendeePromises.push({
+      email: attendee.email,
+      name: attendee.name,
+      timeZone: attendee.timeZone,
+      language: { translate: tAttendee, locale: attendee.locale ?? "en" },
+    });
+  }
+
+  const attendeeList = await Promise.all(attendeePromises);
+
```

Comment:
I don't think this code is utilizing Promise.all() correctly. we are pushing objects and not promises in the array.

Reply:
```
const attendeePromises = [];
for (const attendee of booking.attendees) {
  if (
    attendee.email === reassignedRRHost.email ||
    attendee.email === previousRRHost.email ||
    teamMembers.some((member) => member.email === attendee.email)
  ) {
    continue;
  }
  // Push the promise into the array
  const attendeePromise = getTranslation(attendee.locale ?? "en", "common").then(tAttendee => ({
    email: attendee.email,
    name: attendee.name,
    timeZone: attendee.timeZone,
    language: { translate: tAttendee, locale: attendee.locale ?? "en" },
  }));

  attendeePromises.push(attendeePromise);
}

// Await all promises
const attendeeList = await Promise.all(attendeePromises);
```